### DISCUSSION TITLE
How to do Remote development(SSH) on a Windows Host?

### DISCUSSION THREAD

**Discussion by jon-zu** [microsoft/vscode-cmake-tools#3951] [Q&A]
How do I build a cmake project on a windows host, via ssh remotely. Building the project in a vscode process launched in a vs dev shell works just fine. However since vscode client launches the vscode server on the remote host outside of the dev shell It has to be set up somehow later. I've followed the `tasks.json` setup from [here](https://code.visualstudio.com/docs/cpp/config-msvc#_reusing-your-c-configuration), but It doesn't seem to have any effect.

`tasks.json`
```json
{
    "version": "2.0.0",
    "windows": {
        "options": {
            "shell": {
                "executable": "cmd.exe",
                "args": [
                    "/C",
                    "\"C:\\Program^ Files\\Microsoft^ Visual^ Studio\\2022\\Community\\Common7\\Tools\\VsDevCmd.bat\"",
                    "&&"
                ]
            }
        }
    },
    "tasks": [
        {
            "type": "cmake",
            "label": "CMake: build",
            "command": "build",
            "targets": [
                "all"
            ],
            "preset": "${command:cmake.activeBuildPresetName}",
            "group": "build",
            "problemMatcher": [],
            "detail": "CMake template build task"
        }
    ]
}
```
It's really frustrating that this is so difficult to setup, It should be plug and play.

**Comment by gcampbell-msft** [comment]
@jon-zu We now have a setting which can allow you to always have the Visual Studio Developer environment sourced: cmake.useVsDeveloperEnvironment. 

If you set this to `always` then when you configure and/or build through the typical "CMake: Configure" command, it should work as expected.
